src/go/configgenerator/route_generator.go

```diff
@@ -396,8 +450,12 @@ func makeHttpRouteMatchers(httpRule *httppattern.Pattern) ([]*routepb.RouteMatch
 		}
 	}
 
+	var methodNotAllowedRouteMatchers []*routepb.RouteMatch
 	if httpRule.HttpMethod != httppattern.HttpMethodWildCard {
 		for _, routeMatcher := range routeMatchers {
+			methodUndefinedRouterMatcherMsg := proto.Clone(routeMatcher)
```

Comment:
Do we need a clone? Will anything happen if we don't explicitly clone? I'd like to remove the cast to `*routepb.RouteMatch` in the next line, it seems unsafe.

Reply:
If we don't clone, I will expect the `methodNotAllowedRouteMatcher` shouldn't share the same object with the normal `routeMatcher`, which will add `Method` restriction.